### PR TITLE
Allow referenceId to be optionally supplied and validated as part of transfer payload

### PR DESCRIPTION
**Pull request by iyiolapeter**

### CODE REVIEW

**Comment by raybesiga** [comment]
Thoughts? @ernest-okot

File: test/validate.test.ts [hunk 1]
```diff
@@ -271,9 +271,27 @@ describe("Validate", function() {
   });
 
   describe("validateTransfer", function() {
-    context("when the amount is missing", function() {
+    context("when the referenceId is missing", function() {
       it("throws an error", function() {
         const request = {} as TransferRequest;
+        return expect(validateTransfer(request)).to.be.rejectedWith(
+          "referenceId is required"
```
Comment (ernest-okot) [inline test/validate.test.ts:278]: Doesn't this mean reference id will be required?

Comment (ernest-okot) [reply test/validate.test.ts:278]: Okay, makes sense

**Comment by ernest-okot** [comment]
@iyiolapeter feel free to merge when done

**Pull request merged**